code/modules/spells/aoe_turf/conjure/construct.dm

```diff
@@ -155,5 +161,201 @@
+	playsound(user, 'sound/effects/stonedoor_openclose.ogg', 100, 1)
+	user.throwing = 0
+	AM.layer++
+	var/turf/turf_left = null
+	var/turf/turf_right = null
+	switch	(user.dir)
+		if (SOUTH)
+			turf_left = get_step(AM, EAST)
+			turf_right = get_step(AM, WEST)
+		if (NORTH)
+			turf_left = get_step(AM, WEST)
+			turf_right = get_step(AM, EAST)
+		if (EAST)
+			turf_left = get_step(AM, NORTH)
+			turf_right = get_step(AM, SOUTH)
+		if (WEST)
+			turf_left = get_step(AM, SOUTH)
+			turf_right = get_step(AM, NORTH)
+	if (!turf_left.density && !turf_left.has_dense_content())
+		AM.side1 = new (AM.loc)
+		AM.side1.icon_state += "_side"
+		AM.side1.dir = get_dir(AM, turf_left)
+		spawn (1)
+			AM.side1.forceMove(turf_left)
+	if (!turf_right.density && !turf_right.has_dense_content())
+		AM.side2 = new (AM.loc)
+		AM.side2.icon_state += "_side"
+		AM.side2.dir = get_dir(AM, turf_right)
+		spawn (1)
+			AM.side2.forceMove(turf_right)
+	if (!AM.side1 && AM.side2)
+		var/turf/extra = get_step(turf_right,AM.side2.dir)
+		if (!extra.density && !extra.has_dense_content())
+			spawn (2)
+				AM.side2.forceMove(extra)
+				AM.side1 = new (AM.loc)
+				AM.side1.icon_state += "_mid"
+				AM.side1.dir = get_dir(AM, turf_right)
+				AM.side1.forceMove(turf_right)
+	if (AM.side1 && !AM.side2)
+		var/turf/extra = get_step(turf_left,AM.side1.dir)
+		if (!extra.density && !extra.has_dense_content())
+			spawn (2)
+				AM.side1.forceMove(get_step(turf_left,AM.side1.dir))
+				AM.side2 = new (AM.loc)
+				AM.side2.icon_state += "_mid"
+				AM.side2.dir = get_dir(AM, turf_left)
+				AM.side2.forceMove(turf_left)
+
+
+/obj/effect/forcefield/cult/large
+	desc = "That eerie looking obstacle seems to have been pulled from another dimension through sheer force."
+	name = "Juggerwall"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "juggerwall"
+	light_color = LIGHT_COLOR_RED
+	luminosity = 2
+	invisibility = 0
+	explosion_block = 200
+	var/obj/effect/forcefield/cult/large/side1 = null
+	var/obj/effect/forcefield/cult/large/side2 = null
+
+/obj/effect/forcefield/cult/large/Destroy()
+	if (loc)
+		new /obj/effect/red_afterimage(loc,src)
+	if (side1)
+		qdel(side1)
+	if (side2)
+		qdel(side2)
+	side1 = null
+	side2 = null
+	..()
+	..()
+
+/spell/juggerdash
+	name = "Jugger-Dash"
+	desc = "Charge in a line and knock down anything in your way, even some walls."
+	user_type = USER_TYPE_CULT
+	hud_state = "const_juggdash"
+	override_base = "cult"
+	charge_max = 150
+	spell_flags = 0
+	var/dash_range = 20
+
+/spell/juggerdash/choose_targets(var/mob/user = usr)
+	return list(user)
+
+/spell/juggerdash/cast_check(var/skipcharge = FALSE, var/mob/user = usr)
+	if(user.throwing)
+		return FALSE
+	else
+		return ..()
+
+/spell/juggerdash/cast(var/list/targets, var/mob/user)
+	playsound(user, 'sound/effects/juggerdash.ogg', 100, 1)
+	var/mob/living/simple_animal/construct/armoured/perfect/jugg = user
```

Comment:
What's the deal with these `/perfect` subtypes?  I can't find them defined anywhere.

Reply:
They're the new cult exclusive versions of the 3 constructs.
![image](https://user-images.githubusercontent.com/7573912/46956400-8b441700-d095-11e8-991b-34b5795baed2.png)

Defined in bloodcult_mobs_and_constructs.dm

Reply:
![image](https://user-images.githubusercontent.com/7573912/46956469-b9c1f200-d095-11e8-8e21-6d0731c9b814.png)
You need to use a soul gem to make them, which only cultists can acquire. The old ones are still there: 
![image](https://user-images.githubusercontent.com/7573912/46956504-cd6d5880-d095-11e8-8b95-304a9dc60f15.png)

Reply:
Whoops, I tried to ctrl+f to find them but Github was hiding the file for being too large. All good. 👍